src/browser/engine/webRequest.ts

```diff
@@ -91,34 +92,73 @@ app.on(
         const formattedDetails = recursivelyLowercaseJSONKeys(details);
         const { id, responseheaders } = formattedDetails;
 
+        const headers = new Map<string, string[]>(Object.entries(responseheaders));
+
+        // Override Content Security Policy Header
+        //
+        // `chrome-extension://` is registered as privilegied
+        // with `webFrame.registerURLSchemeAsPrivileged()`
+        // but added iFrames with extension protocol
+        // don't bypass top frame CSP frame-src policy
+        //
+        // ref: https://bugs.chromium.org/p/chromium/issues/detail?id=408932#c35
+        // todo: remove this block since Electron 5 fix the problem
+        //
+        // * Protocol `chrome-extension://` is considered trustworthy
+        // ref: https://w3c.github.io/webappsec-secure-contexts/#is-origin-trustworthy
+        //
+        // * Protocol can bypass secure context check
+        // ref: //src/chrome/common/secure_origin_whitelist.cc
+        // https://cs.chromium.org/chromium/src/chrome/common/secure_origin_whitelist.cc?l=16
+        //
+        // * Protocol added to the Renderer Web Secure Context Safelist
+        // ref: //src/chrome/renderer/chrome_content_renderer_client.cc
+        // https://cs.chromium.org/chromium/src/chrome/renderer/chrome_content_renderer_client.cc?l=428
+        //
+        // * Protocol should bypass CSP check
+        // ref: //src/third_party/blink/renderer/core/frame/csp/content_security_policy.cc?l=706
+        // https://cs.chromium.org/chromium/src/third_party/blink/renderer/core/frame/csp/content_security_policy.cc?l=1557
+        //
+        // * Secure Directive Values
+        // ref: //src/extensions/common/csp_validator.cc
+        // https://cs.chromium.org/chromium/src/extensions/common/csp_validator.cc?l=256
+
+        const cspHeaderKey = 'content-security-policy';
+        const cspPolicyKey = 'frame-src';
+        const cspPolicyDelimiter = ';';
+        const cspDirective: string | undefined = (responseheaders[cspHeaderKey] || [])[0];
+
+        if (cspDirective) {
```

Comment:
Worth adding a little comment explaining what you are actually doing in this block. Or, even better, implement this block in a function with a nice JSDoc in front of it.

Would it make sens to rather use https://github.com/helmetjs/content-security-policy-parser ?

Reply:
Cleaned and used Helmet's parser